packages/@aws-cdk/aws-ec2/lib/instance.ts

```diff
@@ -363,3 +383,32 @@ export class Instance extends Resource implements IInstance {
     }
   }
 }
+
+/**
+ * Synthesize an array of block device mappings from a list of block device
+ *
+ * @param construct the instance/asg construct, used to host any warning
+ * @param blockDevices list of block devices
+ */
+function synthesizeBlockDeviceMappings(construct: Construct, blockDevices: BlockDevice[]): CfnInstance.BlockDeviceMappingProperty[] {
```

Comment:
This is identical to: https://github.com/aws/aws-cdk/blob/48849b5e9cb29228051ef0a8a1f1b41d4daf8c36/packages/%40aws-cdk/aws-ec2/lib/volume.ts#L162

except for the return type.

Could this code be commoned instead of copied, so that when additions are made (ex: adding the encryption properties) then they don't have to be made in all copies?

Reply:
You are right. There is also a difference in the value of `NoDevice`, but let me exercise my TS chops here :)